pkgs/build-support/node/fetch-pnpm-deps/pnpm-config-hook.sh

```diff
@@ -0,0 +1,44 @@
+# shellcheck shell=bash
+
+pnpmConfigHook() {
+    echo "Executing pnpmConfigHook"
+
+    if [ -n "${pnpmRoot-}" ]; then
+      pushd "$pnpmRoot"
+    fi
+
+    if [ -z "${pnpmDeps-}" ]; then
+        echo "Error: 'pnpmDeps' should be set when using pnpmConfigHook."
+        exit 1
+    fi
+
+    echo "Configuring pnpm store"
+
+    export HOME=$(mktemp -d)
+    export STORE_PATH=$(mktemp -d)
+
+    cp -Tr "$pnpmDeps" "$STORE_PATH"
+    chmod -R +w "$STORE_PATH"
+
+    pnpm config set store-dir "$STORE_PATH"
+
+    echo "Installing dependencies"
+
+    if ! pnpm install --offline --frozen-lockfile --ignore-script; then
+        echo
+        echo "ERROR: pnpm failed to install dependencies"
+        echo
+
+        exit 1
+    fi
+
+    patchShebangs node_modules/{*,.*}
+
+    if [ -n "${pnpmRoot-}" ]; then
+      popd
+    fi
+
+    echo "Finished pnpmConfigHook"
+}
+
+postConfigureHooks+=(pnpmConfigHook)
```

Comment:
`npmConfigHook` adds itself to `postPatchHooks`. is there a reason we're not following this example?

Reply:
I personally felt like the configuration step is more fitting, but I am not sure tbh.

Reply:
> I personally felt like the configuration step is more fitting, but I am not sure tbh.

I think it can stay as it is.